### DISCUSSION TITLE
getStaticProps() called at runtime

### DISCUSSION THREAD

**Discussion by mousedown** [vercel/next.js#12575] [Help] [answered]
I've been scratching my head over this for a while. I've tried many angles, please can somebody help? 

I successfully generated .json  for 1000 dynamic pages via `getStaticProps` and `getStaticPaths` however On a full production build  `next build` and  `next start` 
 getStaticProps and therefore the DB is still called on each request. I use both link and router methods. The app ill still SPA, so no issues there. 

`/[type]/[slug]`

Can anybody please suggest a scenario that would cause getStaticProps to be hit at runtime as well as at build? Tomorrow, I will have to go digging through Next codebase to try and figure it out. 

```
export async function getStaticProps(ctx: any) {
  //ctx undefined at build time. 
  console.log('ctx ', ctx.params.slug)
  const sp = await getSingleProduct({ slug: ctx.params.slug });
....
```

**Comment by brianlovin** [comment]
Is this locally, or in production? If locally, I believe Next.js will call `getStaticPaths` and `getStaticProps` every single time. If in prod, I'd be curious to hear as well – do you have the `fallback` key set to `true` in your `getStaticPaths`?

**Comment by mousedown** [comment, marked as answer]
Thanks for the reply, this was both. However, this morning, I've fixed! I was wrong in my initial findings the pages were only hit once per request, indicating the initial lookup was failing. 

Upon looking at the 'with-typescipt' example I can see the generated pages in server are used.  I cleared the cache and generated again and could see the bug. 

`/[type]/[slug]`

but i should have been generating 

`/[type]s/[slug]`

Damn it. I should have spotted that. The cache is working correctly now, super fast pages and great lighthouse score. 😃 

 For anyone else looking for answers, check the generated pages in

`.next/server/static/{hash}/pages`

**Reply by Vadorequest** [reply]
You should mark your question as answered then ;)
